package/MDAnalysis/analysis/polymer.py

```diff
@@ -27,61 +27,158 @@
 
 
 :Author: Richard J. Gowers
-:Year: 2015
+:Year: 2015, 2018
 :Copyright: GNU Public License v3
 
 This module contains various commonly used tools in analysing polymers.
 
+
+Example
+-------
+
+Finding the persistence length of a polymer:
+
+>>> from MDAnalysis.tests.datafiles import TRZ_psf, TRZ
+>>> import MDAnalysis as mda
+>>> from MDAnalysis.analysis import polymer
+>>> u = mda.Universe(TRZ_psf, TRZ)
+>>> # this system is a pure polymer melt, so we can select
+>>> # the chains by using the .fragments attribute
+>>> chains = u.atoms.fragments
+>>> # sort the chains, removing any none backbone atoms
+>>> sorted_chains = [polymer.sort_backbone(f.select_atoms('not name O* H'))
+...                  for f in chains]
+>>> lp = polymer.PersistenceLength(sorted_chains)
+>>> lp = lp.run()
+>>> print('The persistence length is: {}'.format(lp.pl))
+>>> # always check the visualisation of this:
+>>> lp.plot()
+
 """
 from __future__ import division, absolute_import
 from six.moves import range
 
 import numpy as np
 import scipy.optimize
-
+import warnings
 import logging
 
 from .. import NoDataError
+from ..core.groups import requires, AtomGroup
 from ..lib.distances import calc_bonds
 from .base import AnalysisBase
 
 logger = logging.getLogger(__name__)
 
 
+@requires('bonds')
+def sort_backbone(backbone):
+    """Rearrange a linear AtomGroup into backbone order
+
+    Requires that the backbone has bond information,
+    and that only backbone atoms are provided (ie no side
+    chains or hydrogens).
+
+    Parameters
+    ----------
+    backbone : AtomGroup
+      the backbone atoms, not necessarily in order
+
+    Returns
+    -------
+    sorted_backbone : AtomGroup
+      backbone in order, so `sorted_backbone[i]` is bonded to
+      `sorted_backbone[i - 1]` and `sorted_backbone[i + 1]`
+
+
+    .. versionadded:: 0.20.0
+    """
+    if not len(backbone.fragments) == 1:
```

Comment:
Use `if backbone.n_fragments != 1` instead of `len(backbone.fragments)` once #2182 goes through (almost 7 times faster)